### PR TITLE
First pass of SteamAPI

### PR DESCRIPTION
**Pull request by KazWolfe**
Creates a new `Component::SteamApi`, as I'm not really sure where else this stuff belongs.

The SteamAPI code appears to have been at least partially generated via templates/macros, and as such things may not be reflected properly.

### CODE REVIEW

File: FFXIVClientStructs/FFXIV/Component/SteamApi/SteamCallbackBase.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+namespace FFXIVClientStructs.FFXIV.Component.SteamApi;
+
+[StructLayout(LayoutKind.Explicit, Size = Size)]
+public unsafe partial struct SteamCallbackBase {
+    public const int Size = 0x10;
+    
+    [Flags]
+    public enum SteamCallbackFlags : byte {
+        Registered = 1 << 0,
+        GameServer = 2 << 0,  // unused
```
Comment (KazWolfe) [inline FFXIVClientStructs/FFXIV/Component/SteamApi/SteamCallbackBase.cs:10]: Technically correct, but completely wrong.

File: FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs [hunk 2]
```diff
@@ -50,6 +51,14 @@ public unsafe partial struct Framework {
 
     [FieldOffset(0x2BF0)] public GameVersion GameVersion;
 
+    /// <summary>
+    /// Access the Steam API wrapper/interface.
+    /// </summary>
+    /// <remarks>
+    /// The struct backed by this API should not be considered stable. If you use this, you are signing up for API breakage.
+    /// </remarks>
+    [FieldOffset(0x35B8)] public SteamApi SteamApi;
```
Comment (Haselnussbomber) [inline FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:60]: I'm pretty sure this is supposed to be a pointer (see `ffxiv_dx11.exe+95A1D`).
```suggestion
    [FieldOffset(0x35B8)] public SteamApi* SteamApi;
```

Comment (KazWolfe) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:60]: yeah, it is. i was sleep deprived last night.

File: FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+using System.Reflection;
```
Comment (aers) [inline FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs:1]: Was this added by mistake? I don't see a use for it in the other changes in the file.

Comment (KazWolfe) [reply FFXIVClientStructs/FFXIV/Client/System/Framework/Framework.cs]: Yeah, my IDE is overzealous at adding things but terrible at removing them after the fact. I'll fix that.

**Pull request merged**